src/lib/parser_impl.c

```diff
@@ -451,9 +469,153 @@ parser_error_t parser_readPB_SendMsg(const uint8_t *bufferPtr,
+    DEFINE_CONTEXT()
+
+    uint64_t v;
+    while (ctx.offset < ctx.bufferLen && err == parser_ok) {
+        err = _readRawVarint(&ctx, &v);
+        if (err != parser_ok) {
+            return err;
+        }
+
+        switch (FIELD_NUM(v)) {
+            case PBIDX_VOTEMSG_METADATA: {
+                CHECK_NOT_DUPLICATED(votemsg->seen.metadata)
+                READ_ARRAY(votemsg->metadata)
+                break;
+            }
+            case PBIDX_VOTEMSG_PROPOSAL_ID: {
+                CHECK_NOT_DUPLICATED(votemsg->seen.id)
+                READ_ARRAY(votemsg->proposalId)
+                break;
+            }
+            case PBIDX_VOTEMSG_VOTER: {
+                CHECK_NOT_DUPLICATED(votemsg->seen.voter)
+                READ_ARRAY(votemsg->voter)
+                break;
+            }
+            case PBIDX_VOTEMSG_VOTE: {
+                CHECK_NOT_DUPLICATED(votemsg->seen.voteOption)
+                READ_UINT8(votemsg->voteOption)
+                break;
+            }
+            default:
+                // Unknown fields are rejected to avoid malleability
+                return parser_unexpected_field;
+        }
+    }
+
+    err = parser_readPB_Metadata(votemsg->metadataPtr, votemsg->metadataLen, &votemsg->metadata);
+    if (err != parser_ok) return err;
+
     return err;
 }
 
+parser_error_t parser_readPB_UpdateMsg(const uint8_t *bufferPtr,
+                                       uint16_t bufferLen,
+                                       parser_updatemsg_t *updatemsg) {
+    DEFINE_CONTEXT()
+
+    uint64_t v;
+    while (ctx.offset < ctx.bufferLen && err == parser_ok) {
+        err = _readRawVarint(&ctx, &v);
+        if (err != parser_ok) {
+            return err;
+        }
+
+        switch (FIELD_NUM(v)) {
+            case PBIDX_UPDATEMSG_METADATA: {
+                CHECK_NOT_DUPLICATED(updatemsg->seen.metadata)
+                READ_ARRAY(updatemsg->metadata)
+                break;
+            }
+            case PBIDX_UPDATEMSG_ID: {
+                CHECK_NOT_DUPLICATED(updatemsg->seen.id)
+                READ_ARRAY(updatemsg->contractId)
+                break;
+            }
+            case PBIDX_UPDATEMSG_PARTICIPANTS: {
+                // This is a repeated field
+                READ_ARRAY(updatemsg->participants);
+                parser_participant_t participant;
+                parser_ParticipantmsgInit(&participant);
+                err = parser_readPB_Participant(updatemsg->participantsPtr,
+                                                updatemsg->participantsLen,
+                                                &participant);
+                if (err != parser_ok)
+                    return err;
+
+                if(updatemsg->participantsCount > 0) {
+                    //participantsPtr will always point to the first Participant on the array
+                    updatemsg->participantsPtr = (uint8_t *) (updatemsg->participantsPtr - (updatemsg->participantsLen *
```

Comment:
IMO, to avoid the risky pointer arithmetic, I would prefer to keep the original pointer and to do something like again on getItem:
```
                parser_participant_t participant;
                parser_ParticipantmsgInit(&participant);
                err = parser_readPB_Participant(updatemsg->participantsPtr,
                                                updatemsg->participantsLen,
                                                &participant);
```
Even if we have to go through the list multiple times..

Reply:
Yes, it will be safer!. Working on it